test/distributed/tensor/parallel/test_fsdp_2d_parallel.py

```diff
@@ -215,21 +217,56 @@ def _test_2d_e2e_flow(
                 print(name, param_names_2d)
             self.assertTrue(name in param_names_2d)
         self._compare_params(model, model_2d)
-
         if multi_param_group and use_orig_params:
             param_group = [
                 {"params": model.net1.parameters(), "lr": 0.02},
                 {"params": model.net2.parameters(), "lr": 0.15},
             ]
-            optim = torch.optim.Adam(param_group, lr=0.01)
+            if optim_in_backward:
+                _apply_optimizer_in_backward(
+                    optimizer_class=torch.optim.Adam,
+                    params=param_group[0]["params"],
+                    optimizer_kwargs={"lr": param_group[0]["lr"]},
+                )
+                _apply_optimizer_in_backward(
+                    optimizer_class=torch.optim.Adam,
+                    params=param_group[1]["params"],
+                    optimizer_kwargs={"lr": param_group[1]["lr"]},
+                )
```

Comment:
is it possible to avoid duplicate code?

Reply:
it can be deduped by 
```
for param_group_idx in len(param_group):
    _apply_optimizer_in_backward(
                    optimizer_class=torch.optim.Adam,
                    params=param_group[param_group_idx]["params"],
                    optimizer_kwargs={"lr": param_group[param_group_idx]["lr"]},
    )
```

but this doesn't add too much value IMO. lmk what you think

Reply:
Well, there are three places which are doing very similar thing. I do feel we can make it simpler here instead of copy paste six times.

Reply:
I agree that this usage of `_apply_optimizer_in_backward` is pretty verbose and ideally could be made less so.